### ISSUE TITLE
Some implementing questions

### ISSUE THREAD

**Issue by zyqgmzyq** [yuanming-hu/fc4#20]
I have finished implementing your model in tensorflow  without adjustments before you open source. After you open sourse, I'm just referring to the necessary parts and made some changes because your code is complicated ,I have the following results:
25: 0.91    med: 1.75    tri: 1.44    mean: 2.28     75: 3.12      95: 5.79
It's surprising me that the result is higher than the one in your paper. I'm very puzzled. What do I need to do with data before training or anything else is wrong?

**Comment by yuanming-hu** [comment]
Did you do the data augmentation? That's a bit troublesome but essential to performance. You can copy and paste my augmentation code and have a try.
You can also check the confidence values to see if it is learning correctly.

**Comment by zyqgmzyq** [comment]
Yes, but I did not do the color augumentation. And do I need to nonlinearize the image before training?The confidence values is strange. The picture in the second line is  confidence featuremap , the rgb featuremap multiplied before and after the confidence value respectively.
![6001_2 378900](https://user-images.githubusercontent.com/36949881/41198412-78796496-6cad-11e8-8433-bb615322dced.jpg)

**Comment by yuanming-hu** [comment]
It seems that the confidence is not really working here. Also, the patch size seems too small.

**Comment by zyqgmzyq** [comment]
The size is 512*512. Do I need to nonlinearize the image before training? And It's hard to say why the confidence not working.

**Comment by yuanming-hu** [comment]
Could you show me more patches with better content coverage? Actually, this image full of leaves is not very diagnostic.
(non)linearization should not bring you such a big difference.

**Comment by zyqgmzyq** [comment]
![7801_2 448943](https://user-images.githubusercontent.com/36949881/41220899-4b2ce5ce-6d95-11e8-8481-471328fe9d81.jpg)
And this is the test image.
![185_3 176258](https://user-images.githubusercontent.com/36949881/41220935-5ad58e90-6d95-11e8-98df-24a7bcc8c88f.jpg)

**Comment by yuanming-hu** [comment]
everything here seems quite noisy and the network is probably not yet converged. Another cause may be that dropout is not disabled during testing.

**Comment by zyqgmzyq** [comment]
Thank you for your reply!
I did not use dropout during testing。The max epoch is 10000 and the train loss is floating between 1.5 and 2.5. I'm thinking that if I lost something during data preprocessing.

**Comment by yuanming-hu** [comment]
Yeah data augmentation is very important on this tiny dataset. In my fuzzy memory, it may bring down error by 0.3 ...

**Comment by zyqgmzyq** [comment]
Should I do the color augumentation?

**Comment by yuanming-hu** [comment]
Yes, that is important, especially for the SqueezeNet version.

**Comment by zyqgmzyq** [comment]
Get it, Thank you very much.
And in your paper refer that the image is downsample 50% during test, but in your code it cropped to 512*512.

**Comment by yuanming-hu** [comment]
Let me put it like this:
Suppose we have an image of size 3000x1500
We first downsample it to 1500x750.
Then crop a 512x512 "patch" out of it. This is just because we need a fixed size for the CNN during training.
During testing, we still use 1500x750, since it is actually a FCN.

**Comment by zyqgmzyq** [comment]
Thank you!
